### PR TITLE
Clients: using upload api in bin/rucio Fix #670 #468 #501 #29

### LINKED ISSUE #670: Switch bin/rucio upload to uploadclient
Motivation
----------
As the uploadclient is implemented now, the upload code in bin/rucio should use the uploadclient.

### PR DESCRIPTION
**Pull request by TWAtGH**
Clients: using upload api in bin/rucio Fix #670 #468 #501 #29

### CODE REVIEW

**Review by mlassnig — COMMENTED** [review]
looks good to me, but need a second pair of eyes :-) @bari12

File: lib/rucio/tests/test_bin_rucio.py [hunk 1]
```diff
@@ -259,14 +264,16 @@ def test_upload_repeated_file(self):
         remove(tmp_file1)
         remove(tmp_file2)
         remove(tmp_file3)
-        nose.tools.assert_not_equal("File %s:%s successfully uploaded on the storage" % (self.user, tmp_file1[5:]) in out, None)
+        nose.tools.assert_true((self.upload_success_str % tmp_file1_name) in out)
```
Comment (nikmagini) [inline lib/rucio/tests/test_bin_rucio.py:267]: Why was this changed from assert_not_equal to assert_true? Did the logic change for this case? (IIUC trying to upload a file already in the catalog is not an error anymore?)

**Review by nikmagini — COMMENTED** [review]
Looks OK to me - I just have one comment about a change in the test logic

File: lib/rucio/tests/test_bin_rucio.py (hunk 1 above)
Comment (TWAtGH) [reply lib/rucio/tests/test_bin_rucio.py:267]: No, probably a copy paste error. Thanks!

Comment (TWAtGH) [reply lib/rucio/tests/test_bin_rucio.py:267]: But this was wrong anyway wasn't it? Assuming the `... in out` expression returns either True or False, it's always not equal to None.

**Review by bari12 — APPROVED** [review]

File: lib/rucio/tests/test_bin_rucio.py (hunk 1 above)
Comment (nikmagini) [reply lib/rucio/tests/test_bin_rucio.py:267]: I agree that the old test logic didn't make sense because the test was always returning success. IIUC with the new logic, the test will return OK if the file is successfully uploaded after being previously removed - if this is the intended test behavior then it looks fine now.
